:  failed

. [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15205264?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15205264?buildTab=artifacts#/) on master @ [b4891b3542a5cf1da2b22418edbb839937d50560](https://github.com/cockroachdb/cockroach/commits/b4891b3542a5cf1da2b22418edbb839937d50560):


```
//pkg/util/encoding:encoding_test                                     NO STATUS
//pkg/util/encoding/csv:csv_test                                      NO STATUS
//pkg/util/envutil:envutil_test                                       NO STATUS
//pkg/util/errorutil:errorutil_test                                   NO STATUS
//pkg/util/errorutil/unimplemented:unimplemented_test                 NO STATUS
//pkg/util/flagutil:flagutil_test                                     NO STATUS
//pkg/util/fsm:fsm_test                                               NO STATUS
//pkg/util/future:future_test                                         NO STATUS
//pkg/util/fuzzystrmatch:fuzzystrmatch_test                           NO STATUS
//pkg/util/goschedstats:goschedstats_test                             NO STATUS
//pkg/util/grpcutil:grpcutil_test                                     NO STATUS
//pkg/util/grunning:grunning_test                                     NO STATUS
//pkg/util/hlc:hlc_test                                               NO STATUS
//pkg/util/httputil:httputil_test                                     NO STATUS
//pkg/util/humanizeutil:humanizeutil_test                             NO STATUS
//pkg/util/interval:interval_test                                     NO STATUS
//pkg/util/interval/generic:generic_test                              NO STATUS
//pkg/util/intsets:intsets_test                                       NO STATUS
//pkg/util/ipaddr:ipaddr_test                                         NO STATUS
//pkg/util/json:json_disallowed_imports_test                          NO STATUS
//pkg/util/json:json_test                                             NO STATUS
//pkg/util/json/tokenizer:tokenizer_test                              NO STATUS
//pkg/util/limit:limit_test                                           NO STATUS
//pkg/util/log:log_test                                               NO STATUS
//pkg/util/log/eventpb:eventpb_test                                   NO STATUS
//pkg/util/log/logconfig:logconfig_test                               NO STATUS
//pkg/util/log/logcrash:logcrash_test                                 NO STATUS
//pkg/util/log/logmetrics:logmetrics_test                             NO STATUS
//pkg/util/log/testshout:testshout_test                               NO STATUS
//pkg/util/metamorphic:metamorphic_test                               NO STATUS
//pkg/util/metric:metric_test                                         NO STATUS
//pkg/util/metric/aggmetric:aggmetric_test                            NO STATUS
//pkg/util/mon:mon_test                                               NO STATUS
//pkg/util/netutil:netutil_test                                       NO STATUS
//pkg/util/netutil/addr:addr_test                                     NO STATUS
//pkg/util/optional:optional_test                                     NO STATUS
//pkg/util/parquet:parquet_test                                       NO STATUS
//pkg/util/pprofutil:pprofutil_test                                   NO STATUS
//pkg/util/pretty:pretty_test                                         NO STATUS
//pkg/util/protoutil:protoutil_test                                   NO STATUS
//pkg/util/quantile:quantile_test                                     NO STATUS
//pkg/util/queue:queue_test                                           NO STATUS
//pkg/util/quotapool:quotapool_test                                   NO STATUS
//pkg/util/randident:randident_test                                   NO STATUS
//pkg/util/randutil:randutil_test                                     NO STATUS
//pkg/util/rangedesc:rangedesc_test                                   NO STATUS
//pkg/util/retry:retry_test                                           NO STATUS
//pkg/util/ring:ring_test                                             NO STATUS
//pkg/util/schedulerlatency:schedulerlatency_test                 
, stderr: 
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38621

```
pkg/roachprod/promhelperclient/client_test.go:146:12: unchecked error (errcheck)
pkg/roachprod/promhelperclient/client_test.go:147:12: unchecked error (errcheck)
pkg/roachprod/promhelperclient/client_test.go:157:12: unchecked error (errcheck)
pkg/roachprod/promhelperclient/client_test.go:158:12: unchecked error (errcheck)
pkg/roachprod/promhelperclient/client_test.go:168:12: unchecked error (errcheck)
pkg/roachprod/promhelperclient/client_test.go:169:12: unchecked error (errcheck)
```

--- a/pkg/roachprod/promhelperclient/client_test.go
+++ b/pkg/roachprod/promhelperclient/client_test.go
@@ -143,8 +143,10 @@ func Test_getToken(t *testing.T) {
 		require.Empty(t, token)
 	})
 	t.Run("invalid credentials", func(t *testing.T) {
-		os.Setenv(ServiceAccountJson, "{}")
-		os.Setenv(ServiceAccountAudience, "dummy_audience")
+		err := os.Setenv(ServiceAccountJson, "{}")
+		require.Nil(t, err)
+		err = os.Setenv(ServiceAccountAudience, "dummy_audience")
+		require.Nil(t, err)
 		c.newTokenSource = func(ctx context.Context, audience string, opts ...idtoken.ClientOption) (oauth2.TokenSource, error) {
 			return nil, fmt.Errorf("invalid")
 		}
@@ -154,8 +156,10 @@ func Test_getToken(t *testing.T) {
 		require.Equal(t, "error creating GCS oauth token source from specified credential: invalid", err.Error())
 	})
 	t.Run("invalid token", func(t *testing.T) {
-		os.Setenv(ServiceAccountJson, "{}")
-		os.Setenv(ServiceAccountAudience, "dummy_audience")
+		err := os.Setenv(ServiceAccountJson, "{}")
+		require.Nil(t, err)
+		err = os.Setenv(ServiceAccountAudience, "dummy_audience")
+		require.Nil(t, err)
 		c.newTokenSource = func(ctx context.Context, audience string, opts ...idtoken.ClientOption) (oauth2.TokenSource, error) {
 			return &mockToken{token: "", err: fmt.Errorf("failed")}, nil
 		}
@@ -165,8 +169,10 @@ func Test_getToken(t *testing.T) {
 		require.Equal(t, "error getting identity token: failed", err.Error())
 	})
 	t.Run("success", func(t *testing.T) {
-		os.Setenv(ServiceAccountJson, "{}")
-		os.Setenv(ServiceAccountAudience, "dummy_audience")
+		err := os.Setenv(ServiceAccountJson, "{}")
+		require.Nil(t, err)
+		err = os.Setenv(ServiceAccountAudience, "dummy_audience")
+		require.Nil(t, err)
 		c.newTokenSource = func(ctx context.Context, audience string, opts ...idtoken.ClientOption) (oauth2.TokenSource, error) {
 			return &mockToken{token: "token"}, nil
 		}

